drivers/network/dd/pcnet/pcnet.c

```diff
@@ -140,7 +140,7 @@ MiniportHandleInterrupt(
 
               RtlZeroMemory(Descriptor, sizeof(RECEIVE_DESCRIPTOR));
               Descriptor->RBADR =
-                  (ULONG)(Adapter->ReceiveBufferPtrPhys + Adapter->CurrentReceiveDescriptorIndex * BUFFER_SIZE);
+                  (ULONG_PTR)(Adapter->ReceiveBufferPtrPhys + Adapter->CurrentReceiveDescriptorIndex * BUFFER_SIZE);
```

Comment:
Should this also keep the additional cast for truncation?
(actually ReceiveBufferPtrPhys shouldn't be a pointer at all, I don't know why it is)